xpdan/db_utils.py

```diff
@@ -0,0 +1,100 @@
+from pprint import pprint
+from .utils import _timestampstr
+
+
+def need_name_here(hdrs, key, verbose=True):
+    """Build a dictionary where the keys are the unique values for the given
+    key and the values are the positions of the headers for that key
+
+    Parameters
+    ----------
+    hdrs: list of Header objects
+        The headers to be searched
+    key: str
+        The key to be searched for
+    verbose: bool, optional
+        If true prints the results. Defaults to True
+
+    Returns
+    -------
+    dict:
+        The dictionary of unique values and positions
+    """
+    d = {}
+    for i, hdr in enumerate(hdrs):
+        if key in hdr.start.keys():
+            if hdr.start[key] in d.keys():
+                d[hdr.start[key]].append(i)
+            else:
+                d[hdr.start[key]] = [i]
+    if verbose:
+        pprint(d)
+    return d
+
+
+def scan_diff(hdrs, verbose=True):
+    """Get the metadata differences between scans
+
+    Parameters
+    ----------
+    hdrs: list of Header objects
+        The headers to be diffed
+    verbose: bool, optional
+        If true prints the results. Defaults to True
+
+    Returns
+    -------
+    dict:
+        The dictionary of keys with different values across the scans.
+        The values are the results for each header.
+    """
+    keys = set([k for hdr in hdrs for k in hdr.start.keys()])
+    kv = {}
+    for k in keys:
+        v = [hdr.start[k] for hdr in hdrs if k in hdr.start.keys()]
+        if len(set(v)) != 1:
+           kv[k] = v
+    if verbose:
+        pprint(kv)
+    return kv
+
+
+def scan_headlines(hdrs, fields=None, verbose=True):
+    """Provide one line summaries of headers
+
+    Parameters
+    ----------
+    hdrs: list of headers
+        The headers from the databroker
+    fields: list of str, optional
+        The fields to be included in the summary, if None use
+        `['sample_name', 'temperature', 'diff_x', 'diff_y', 'eurotherm']`
+        defaults to None
+    verbose: bool, optional
+        If True print the summary
+
+    Returns
+    -------
+    list:
+        List of summary strings
+    """
+    if fields is None:
+        fields = ['sample_name', 'temperature', 'diff_x', 'diff_y', 'eurotherm']
```

Comment:
'sample_name', 'temperature', 'diff_x', 'diff_y', 'eurotherm','cs700' (not sure, but we want the temperature returned by the cryostream controller), count time, scan_type, start_temp, end_temp, temp_step, exposure, pause

note to self: make an issue sot hat this is controlled by global/config later